### PR TITLE
flames remove snow ! 

### PR DESCRIPTION
**Pull request by cuberound**
# About the pull request

makes snow scortchable and implements behavior for it ( thank you kugamo for making the LV grass scortchable so that the basework for it exists). intensity numbers might need tweeking if maintainers feel like it, but other than that it works.

# Explain why it's good for the game

removing snow is pain in ass and even more so for marines, explosions can already remove snow and it only makes sance for flames to melt the snow. only bluefire can remove the deapest snow in one use, green and red flames remove abut two layers

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
balance: flames melt the snow 
/:cl:

### CODE REVIEW

**Review by Zonespace27 — DISMISSED** [review]

File: code/game/turfs/auto_turf.dm [hunk 1]
```diff
@@ -75,6 +75,20 @@
 			if(bleed_layer)
 				addtimer(CALLBACK(src, PROC_REF(changing_layer), 0), 1)
 
+/turf/open/auto_turf/scorch(heat_level)
+	if(bleed_layer == 0)
```
Comment (Drulikar) [inline code/game/turfs/auto_turf.dm:79]:
```suggestion
	if(bleed_layer <= 0)
```

File: code/game/turfs/auto_turf.dm [hunk 2]
```diff
@@ -75,6 +75,20 @@
...
+		return
+	switch(heat_level)
+		if(1 to 19)
+			var/new_bleed_layer = min(0, bleed_layer - 1)
+			addtimer(CALLBACK(src, PROC_REF(changing_layer), new_bleed_layer), 1)
+
```
Comment (Drulikar) [inline code/game/turfs/auto_turf.dm:85]:
```suggestion
```

File: code/game/turfs/snow.dm [hunk 3]
```diff
@@ -132,6 +133,23 @@
...
+		if(1 to 19)
+			bleed_layer--
+			update_icon(TRUE, FALSE)
+
+		if(20 to 39)
+			bleed_layer = max(bleed_layer - 2, 0)
+			update_icon(TRUE, FALSE)
+		if(40 to INFINITY)
+			bleed_layer = 0
+			update_icon(TRUE, FALSE)
```
Comment (Drulikar) [inline code/game/turfs/snow.dm:150]:
```suggestion
			update_icon(update_full = TRUE, skip_sides = FALSE)
		if(20 to 39)
			bleed_layer = max(bleed_layer - 2, 0)
			update_icon(update_full = TRUE, skip_sides = FALSE)
		if(40 to INFINITY)
			bleed_layer = 0
			update_icon(update_full = TRUE, skip_sides = FALSE)
```

File: code/game/turfs/snow.dm [hunk 4]
```diff
@@ -132,6 +133,23 @@
 				bleed_layer = 0
 				update_icon(1, 0)
 
+//Flames act
+/turf/open/snow/scorch(heat_level)
+	if(bleed_layer == 0)
```
[the remark this answers is not in the document]
Comment (Drulikar) [reply code/game/turfs/snow.dm:138]: Unresolved

**Review by Drulikar — CHANGES_REQUESTED** [review]

**Review by Drulikar — APPROVED** [review]
Works. Though I will note it may make sense if adjacent snow tiles melted a layer too - but it doesn't have to. Also the scorch seems to only happen once so if its a deep enough patch of snow and a low enough flame, you'll still have snow remain despite fire continuing to burn on it for a while.

**Pull request merged**